crsqlite_extension/src/storage.c

```diff
@@ -0,0 +1,106 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdarg.h>
+
+#include <sqlite3ext.h>
+
+#ifndef sqlite3_api
+SQLITE_EXTENSION_INIT3
+#endif
+
+#define COLUMN_NAME 1
+#define COLUMN_TYPE 2
+#define COLUMN_NOT_NULL 3
+#define COLUMN_DFLT_VAL 4
+#define COLUMN_PK 5
+
+ /*
+** Construct one or more SQL statements from the format string given
+** and then evaluate those statements. The success code is written
+** into *pRc.
+**
+** If *pRc is initially non-zero then this routine is a no-op.
+*/
+static void storageDbExec(
+  int *pRc,              /* Success code */
+  sqlite3 *db,           /* Database in which to run SQL */
+  const char *zFormat,   /* Format string for SQL */
+  ...                    /* Arguments to the format string */
+){
+  va_list ap;
+  char *zSql;
+  if( *pRc ) return;
```

Comment:
weird. Why would a return code already exist when the function is called?

Reply:
The result code is set as a pointer because its a variable argument function. This should be "if ( *pRc != SQLITE_OK ) return;" for clarity